### ISSUE TITLE
Rails 6.1.4 system tests for scaffolding fail out of the box with rack_test

### ISSUE THREAD

**Issue by profsmucker** [rails/rails#42780]
### Steps to reproduce

```
rails new .
```
edit test/application_system_test_case.rb to be driven_by :rack_test:
```
require "test_helper"

class ApplicationSystemTestCase < ActionDispatch::SystemTestCase
  driven_by :rack_test
end
```
edit Gemfile to include the gem rexml (this is another "out of box" issue that blocks testing with Ruby 3.0):
```
gem 'rexml'
```
then:
```
bundle install
rails generate scaffold post msg:string
rails db:create db:migrate
rails test:system
```

### Expected behavior
Tests run successfully.

### Actual behavior
For any helper methods that use the url form, e.g. posts_url, get this error (for example):

```
PostsTest#test_visiting_the_index:
ArgumentError: Missing host to link to! Please provide the :host parameter, set default_url_options[:host], or set :only_path to true
    test/system/posts_test.rb:10:in `block in <class:PostsTest>'
```
This is caused by this change in Rails: https://github.com/rails/rails/commit/d415eb4f6d6bb24b78b968ae28c22bb7e1721285#diff-e51da1e561e29eb5bb869aaa39ef9174ec9e721838d7bd524a98bb4d3c48e6b5

System testing works as expected if we set:
```
Capybara.app_host = 'http://www.example.com'
```
or
```
Capybara.app_host = 'http://127.0.0.1'
```
Without this, the default app_host is nil and the system test fails to work out of the box.

### System configuration
**Rails version**: Rails 6.1.4

**Ruby version**: ruby 3.0.1p64 (2021-04-05 revision 0fb782ee38) [x86_64-linux]

**Comment by rafaelfranca** [comment]
What does `Capybara.current_session.server_url` return?

**Comment by profsmucker** [comment]
nil:
```
[5, 14] in /home/codio/workspace/hello/test/system/posts_test.rb
    5:     @post = posts(:one)
    6:   end
    7: 
    8:   test "visiting the index" do
    9:     byebug
=> 10:     visit posts_url
   11:     assert_selector "h1", text: "Posts"
   12:   end
   13: 
   14:   test "creating a Post" do
(byebug) Capybara.current_session.server_url
nil
(byebug) Capybara.app_host
nil
(byebug) n

[196, 205] in /usr/share/rvm/gems/ruby-3.0.1/gems/minitest-5.14.4/lib/minitest/test.rb
   196:     rescue *PASSTHROUGH_EXCEPTIONS
   197:       raise
   198:     rescue Assertion => e
   199:       self.failures << e
   200:     rescue Exception => e
=> 201:       self.failures << UnexpectedError.new(e)
   202:     end
   203: 
   204:     def with_info_handler &block # :nodoc:
   205:       t0 = Minitest.clock_time
(byebug) e
#<ArgumentError: Missing host to link to! Please provide the :host parameter, set default_url_options[:host], or set :only_path to true>
```

**Comment by profsmucker** [comment]
I've updated my original issue to note that you need to add gem rexml to the Gemfile or else you cannot run any tests with Ruby 3.0.  This is another "out of the box" issue with Rails 6.1.4 and Ruby 3.0.

**Comment by mwallba** [comment]
After researching this a bit it looks like there are indeed two separate issues at play here:

1. The missing app_host for Capybara as described above
2. The REXML load issue

The latter is caused by the fact that Ruby 3 extracted REXML into a bundled gem: [Feature #16485 Make rexml, rss to the bundled gems on ruby-lang.org Tracker](https://bugs.ruby-lang.org/issues/16485)

This issue popped up in the bootsnap repo too: https://github.com/Shopify/bootsnap/issues/325 but is in fact caused by selenium-webdriver depending on rexml and has already been discussed in #41502. 

As mentioned there, we could pin the current selenium-webdriver beta (which explicitly requires rexml) or we require rexml in the Gemfile ourselves as pointed out by @profsmucker. Unfortunately there hasn't been a release of selenium-webdriver 4 yet and judging by https://github.com/SeleniumHQ/selenium/pull/9002 there will be no backport of the fix to version 3

@rafaelfranca Do you have a preference on how to proceed with the REXML issue and should we maybe continue the discussion for that issue in #41502 or a new, separate issue?

**Comment by p8** [comment]
This is still reproducible on main (7d8dd7457fbcaa4862099b4ca28f073669460ef8)

**Comment by rafaelfranca** [comment]
The rexml issue should be fixed by b06a6a9d25.

**Comment by profsmucker** [comment]
This is still broken in 6.1.4.4, but the rexml issue is now fixed.
